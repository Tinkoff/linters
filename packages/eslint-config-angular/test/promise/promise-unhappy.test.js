import ESlint from 'eslint';
import path from 'path';

describe('promise / unhappy', () => {
  const cli = new ESlint.CLIEngine({
    cwd: path.join(__dirname, '..'),
    useEslintrc: false,
    baseConfig: {
      extends: ['../promise'],
      parserOptions: {
        project: 'tsconfig.json',
      },
    },
  });

  it('unhappy', () => {
    const codeframe = cli.getFormatter('codeframe');
    const report = cli.executeOnFiles([
      path.join(__dirname, './__fixtures__/promise-unhappy.fixture.ts'),
    ]);

    expect(codeframe(report.results)).toMatchSnapshot();
  });
});
