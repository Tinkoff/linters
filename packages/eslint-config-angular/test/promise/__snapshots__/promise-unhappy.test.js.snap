// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`promise / unhappy unhappy 1`] = `
"error: Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the \`void\` operator (@typescript-eslint/no-floating-promises) at packages/eslint-config-angular/test/promise/__fixtures__/promise-unhappy.fixture.ts:2:1:
  1 | const unhappyBar = new Promise(() => {});
> 2 | unhappyBar;
    | ^
  3 | const unhappyFoo = void unhappyBar;
  4 | 
  5 | function unhappyBaz() {


error: Expected 'undefined' and instead saw 'void' (no-void) at packages/eslint-config-angular/test/promise/__fixtures__/promise-unhappy.fixture.ts:3:20:
  1 | const unhappyBar = new Promise(() => {});
  2 | unhappyBar;
> 3 | const unhappyFoo = void unhappyBar;
    |                    ^
  4 | 
  5 | function unhappyBaz() {
  6 |   return void 0;


error: Expected 'undefined' and instead saw 'void' (no-void) at packages/eslint-config-angular/test/promise/__fixtures__/promise-unhappy.fixture.ts:6:10:
  4 | 
  5 | function unhappyBaz() {
> 6 |   return void 0;
    |          ^
  7 | }
  8 | unhappyBaz()
  9 | 


3 errors found."
`;
